lib/location/method_gnss.c

```diff
@@ -878,8 +877,11 @@ int method_gnss_location_get(const struct location_method_config *config)
 	}
 #endif
 #if defined(CONFIG_NRF_CLOUD_AGPS) || defined(CONFIG_NRF_CLOUD_PGPS)
-	/* Trigger GNSS to send NRF_MODEM_GNSS_EVT_AGPS_REQ event if assistance data is needed. */
 	k_work_submit_to_queue(location_core_work_queue_get(), &method_gnss_agps_req_work);
+	/* Sleep for a while before submitting the next work, otherwise A-GPS data may not be
+	 * downloaded before GNSS is started.
```

Comment:
I had to check the code to see why this sleep is needed despite this comment. In Teams you explained this more thoroughly so I'd add more stuff in here what happens why this sleep is required.
E.g., we start GNSS, wait for `NRF_MODEM_GNSS_EVT_AGPS_REQ` event and put the AGPS request sending to the cloud into the work queue. This all has to occur before submitting `method_gnss_start_work` into work queue.

Reply:
Ok, I'll add more details.